packages/dos-donts-block/src/components/ImageComponent.tsx

```diff
@@ -0,0 +1,154 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { Button, ButtonEmphasis, IconArrowCircleUp, IconImageStack, IconPlus32, LoadingCircle } from '@frontify/fondue';
+import { joinClassNames, radiusStyleMap, toRgbaString } from '@frontify/guideline-blocks-shared';
+import { useMemo } from 'react';
+import { DoDontImageHeight, IMAGE_HEIGHT_VALUES, ImageComponentProps, ImageFitChoice } from '../types';
+
+const ImageComponent = ({
+    isEditing,
+    src,
+    columns,
+    onAssetChooseClick,
+    onUploadClick,
+    isUploadLoading,
+    customImageHeightValue,
+    imageDisplay,
+    imageHeightChoice,
+    isCustomImageHeight,
+    hasStrikethrough,
+    draggableProps,
+    isDragging,
+    backgroundColor,
+    borderColor,
+    borderStyle,
+    borderWidth,
+    hasBackground,
+    hasBorder,
+    hasRadius,
+    radiusChoice,
+    radiusValue,
+}: ImageComponentProps) => {
+    const imageHeight = useMemo(() => {
+        if (isCustomImageHeight) {
+            return customImageHeightValue;
+        } else {
+            return IMAGE_HEIGHT_VALUES[imageHeightChoice];
+        }
+    }, [customImageHeightValue, imageHeightChoice, isCustomImageHeight]);
```

Comment:
Why is the `useMemo` needed? I think this could simplified to the following (also for all the other `useMemo` below)
```suggestion
    const imageHeight = isCustomImageHeight ? customImageHeightValue : IMAGE_HEIGHT_VALUES[imageHeightChoice];
```

Reply:
`useMemo` should only be used when you want to explicitly avoid rerendering of a component